### PR TITLE
feat: configure interfaces concurrently

### PR DESCRIPTION
**Pull request by andrewrynhard**
This uses a wait group to configure interfaces concurrently.

### CODE REVIEW

File: internal/app/networkd/pkg/networkd/networkd.go [hunk 1]
```diff
@@ -80,33 +81,41 @@ func (n *Networkd) Configure(ifaces ...*nic.NetworkInterface) error {
...
-		for _, method := range iface.AddressMethod {
-			log.Printf("configuring %s addressing for %s\n", method.Name(), iface.Name)
+	wg.Add(len(ifaces))
 
-			if err = n.configureInterface(method); err != nil {
-				// Treat as non fatal error when failing to configure an interface
-				log.Println(err)
-				continue
+	for _, iface := range ifaces {
+		go func(i *nic.NetworkInterface) {
```
Comment (smira) [inline internal/app/networkd/pkg/networkd/networkd.go]: no `defer wg.Done()` ?

I bet this should hang forever...

Comment (andrewrynhard) [reply internal/app/networkd/pkg/networkd/networkd.go]: whoops

Comment (andrewrynhard) [reply internal/app/networkd/pkg/networkd/networkd.go]: Fixed

File: internal/app/networkd/pkg/networkd/networkd.go [hunk 2]
```diff
@@ -80,33 +81,42 @@ func (n *Networkd) Configure(ifaces ...*nic.NetworkInterface) error {
...
+					log.Println(err)
+					return
+				}
+
+				if !method.Valid() {
+					return
+				}
+
+				// Aggregate a list of DNS servers/resolvers
+				resolvers = append(resolvers, method.Resolvers()...)
```
Comment (smira) [inline internal/app/networkd/pkg/networkd/networkd.go]: this should have some locking as it's data race

Comment (andrewrynhard) [reply internal/app/networkd/pkg/networkd/networkd.go]: good catch

Comment (smira) [reply internal/app/networkd/pkg/networkd/networkd.go]: I'm not sure if that scope is right? should it be local variable? 

`resolvers` is a local variable in this function

Comment (andrewrynhard) [reply internal/app/networkd/pkg/networkd/networkd.go]: 🤦‍♂ I'm tired today. Long night with the baby.

File: internal/app/networkd/pkg/networkd/networkd.go [hunk 3]
```diff
@@ -78,35 +79,47 @@ func (n *Networkd) Configure(ifaces ...*nic.NetworkInterface) error {
...
+					log.Println(err)
+					return
+				}
+
+				if !method.Valid() {
+					return
+				}
+
+				// Aggregate a list of DNS servers/resolvers
+				mu.Lock()
```
Comment (bradbeam) [inline internal/app/networkd/pkg/networkd/networkd.go]: unsure if it's better to use a lock here or just kick off another loop afterwards
```
for _, iface := range ifaces {
  go func(i *nic.NetworkInterface) { ... }
}
wg.Wait()
var resolvers []string
for _, iface := range ifaces {
  resolvers = append(resolvers, method.Resolvers()...)
}

Comment (andrewrynhard) [reply internal/app/networkd/pkg/networkd/networkd.go]: One for loop better than two maybe? IDK. We could probably go either way but int the end one doesn't have something over the other IMO.

**Pull request merged**